arches_for_science/media/js/views/components/workflows/project-report-workflow/download-report.js

```diff
@@ -37,6 +46,24 @@ define([
         getProjectName();
 
         const generateReport = async(template) => {
+            const relatedObjects = await getRelatedResources(projectId);
+            const collections = relatedObjects.related_resources.filter(rr => rr.graph_id == collectionGraphId);
+            const allPhysicalThingsResponse = collections.map(async(collection) => {
+                const collectionRelatedResources = await getRelatedResources(collection.resourceinstanceid);
+                return collectionRelatedResources?.related_resources.filter(rr => rr.graph_id == physicalThingGraphId);
+            });
+
+            self.physicalThings = [].concat(...(await Promise.all(allPhysicalThingsResponse))).filter(res => {
```

Comment:
a cosmetic suggestion at your option
```suggestion
            self.physicalThings = [...(await Promise.all(allPhysicalThingsResponse))].filter(res => {
```

Reply:
I believe thus is intentional since the sample is a real object unlike the areas.
But the requirements are still being refined, so it may be changed in the future, as well as the above (the request on collection) For example, do all the physical things to be included or just the one that the user selected...)
This should be good for now.

Reply:
The Promise.all returns an array of arrays.
If the response spreads in [], we will end up having an array of arrays again (The above 2nd cod block).
If the response spreads and concats then we will get an array of the objects. (I think that's what we want)

Reply:
I swapped in the suggestion and it still returns an array of objects. I think it's because Promise.all() returns an array of objects/resolved promises.

Fine to leave this as is, I just tend to ask questions like this when I'm not sure I have the right understanding of what's going on :-)

Reply:
~Oh! The filter is chained to the concat, sorry. Now I see what it's doing.~

Reply:
Thanks for your patience while I worked through this pattern; I'd never seen this strategy for reducing dimensionality before through concat!